debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PgOid.java

```diff
@@ -58,4 +80,43 @@ protected static int jdbcColumnToOid(Column column) {
             return Oid.UNSPECIFIED;
         }
     }
+
+    /**
+     * Converts a type name in long (readable) format like <code>boolean</code> to s standard
+     * data type name like <code>bool</code>.
+     * 
+     * @param typeName - a type name in long format 
+     * @return - the type name in standardized format
+     */
+    public static String normalizeTypeName(String typeName) {
+        return LONG_TYPE_NAMES.getOrDefault(typeName, typeName);
+    }
+
+    /**
+     * JDBC metadata are different for some of the unbounded types from those coming via decoder.
+     * This method sets the type constraints to the values provided by JDBC metadata. 
+     * 
+     * @param column - a column coming from decoder
+     * @param columnEditor - the JDBC counterpart of the column
+     */
+    public static void reconcileJdbcOidTypeConstraints(final ReplicationMessage.Column column,
+            final ColumnEditor columnEditor) {
+        switch (column.getTypeName()) {
```

Comment:
Ok for now, but shouldn't we fix this in DecoderBufs?

Reply:
Good question. I was thinking about having most of type handling centralized in one place - easier to mainatin when new types are added.

Reply:
I think in this case it'd really make more sense to adjust that in DecoderBufs. But can be done separately.